lib/ansible/modules/cloud/azure/azure_rm_networkinterface.py

```diff
@@ -77,58 +76,90 @@
             - Windows
             - Linux
         default: Linux
-        required: false
     private_ip_address:
         description:
-            - Valid IPv4 address that falls within the specified subnet.
-        required: false
+            - (Deprecate) Valid IPv4 address that falls within the specified subnet.
+            - This option will be deprecated in 2.9, use I(ip_configurations) instead.
     private_ip_allocation_method:
         description:
-            - "Specify whether or not the assigned IP address is permanent. NOTE: when creating a network interface
+            - "(Deprecate) Specify whether or not the assigned IP address is permanent. NOTE: when creating a network interface
               specifying a value of 'Static' requires that a private_ip_address value be provided. You can update
               the allocation method to 'Static' after a dynamic private ip address has been assigned."
+            - This option will be deprecated in 2.9, use I(ip_configurations) instead.
         default: Dynamic
         choices:
             - Dynamic
             - Static
-        required: false
     public_ip:
```

Comment:
Why is there no `public_ip` in `ip_configurations`?

Reply:
Since the flatten parameters are compressed as one item in a list, and I need to compare the no-order list when updating.

The previous behavior will create the default name for the `public_ip_address_name` for the public ip if `public_ip` flag is set to true. And set the only one as `primary` one.
 
But going to list, it should specify a `primary` one among the list. Also I cannot set the default name for public ip, since it is difficult to know the updating. So I make the `primary` and `public_ip_address_name` required. As the `public_ip_address_name` is required if it needs public ip, the flag `public_ip` is no need here to identify whether it is public.

Reply:
@jborean93